packages/zchain-core/src/lib/zid.ts

```diff
@@ -0,0 +1,50 @@
+import PeerId from "peer-id";
+import fs from "fs";
+import path from "path";
+import { getPathFromDirRecursive } from "./files";
+
+const PEER_ID_DIR = "ids";
+const jsonExt = ".json";
+
+/**
+ * Class representing zchain id (a persistent peer id accross zchain nodes)
+ */
+export class ZID {
+  public name: string | undefined; // maybe we can use a name associated with peer id?
```

Comment:
Yeah, there will be no way to ensure it is unique. We will add a verification later with a zNA attribute (ZERO Name Address -> e.g., '0://wilder.meow'), that connects and confirm they hold a valid zNA. This will be optional.

Reply:
hmm, alright